shared/js/sk_key_utils.js

```diff
@@ -0,0 +1,367 @@
+    }
+
+    /**
+     * Used to create soft key container
+     * @example
+     * createSoftKeyContainer();
+     *
+     * @param N/A
+     * @return void 0
+     */
+    function createSoftKeyContainer() {
+      var sk_button_container = null;
+
+      // Parent container
+      sk_container = document.createElement('div');
+      sk_container.dataset.subtype = 'menu-parent';
+      sk_container.tabIndex = -1;
+      sk_container.setAttribute('role', 'dialog');
+
+      sk_button_container = document.createElement('div');
+      sk_button_container.classList.add('sk-button-container');
+      sk_container.appendChild(sk_button_container);
+
+      document.body.appendChild(sk_container);
+    }
+
+    /**
+     * Used to create soft key buttons
+     * @example
+     * createSoftKeyButtons(element, 'LSK');
+     *
+     * @param {object, string}
+     *        key: Key which will be pressed
+     *        keyType: Key type ie. LSK, RSK, CSK. 
+     *                 Required for adding styles
+     * @return void 0
+     */
+    function createSoftKeyButtons(key, keyType) {
+      var button = document.createElement('div');
+      button.textContent = key.textContent;
+      button.classList.add('sk-buttons', keyType);
+      sk_container
+              .querySelector('.sk-button-container')
+              .appendChild(button);
+
+      // Hide the buttons for Non-Touch UI
+      key.classList.add('sk-hide-button');
+    }
+
+    /**
+     * Used to handle button press
+     * @example
+     * handleSoftKey();
+     *
+     * @param N/A
+     * @return void 0
+     */
+    function handleSoftKey() {
+      // Check if menu open
+      if (!sk_options_menu.open) {
+        // Get context menu
+        var contextMenu = this.getAttribute('contextmenu');
+        var menuItems = null;
+
+        // Show options for context menu
+        if (contextMenu) {
+          var menu = document.getElementById(contextMenu);
+          if (!menu || !menu.children) {
+            throw Error('Menu not present');
+          }
+
+          menuItems = [].slice.call(menu.children);
+
+          // Show options menu
+          // TODO: Handle cancel event if menu is visible.
+          showOptionsMenu({
+            items: menuItems,
+            header: menu.getAttribute('header')
+          });
+        }
+        else {
+          // TODO: Do we need to raise events from here or
+          // individual developer will handle LSK,CSK,RSK press ?
+        }
+      }
+      else {
+        // Binded to LSK
+        if (this.getAttribute('accesskey') ===
+                sk_options_menu.cancelKey) {
+
+          // Close the menu
+          hideOptionsMenu();
+        }
+      }
+    }
+
+    /**
+     * Used to create options menu
+     * @example
+     * showOptionsMenu({items:items,header:'header'});
+     *
+     * @param {object} options: Pass options
+     * @return void 0
+     */
+    function showOptionsMenu(options) {
```

Comment:
I'm ignoring this bit for now. It might be worth having a separate module for displaying menus, if that is possible.